src/vec_bcd_ppc.h

```diff
@@ -880,9 +2230,171 @@ vec_rdxct10e32q (vui64_t vra)
+	result = (vb128_t) vec_splat_s8(0);
+    }
+  else
+    result = (vb128_t) vec_splat_s8(-1);
+#endif
+  return (result);
+}
+
+/*! \brief Vector Set Bool from Signed BCD Quadword.
+ *
+ *  If the quadword's sign nibble is 0xB or 0xD then return a vector
+ *  bool __int128 that is all '1's. Otherwise if the sign nibble is
+ *  0xA, 0xC, 0xE, or 0xF then return all '0's.
+ *
+ *  /note For _ARCH_PWR7 and earlier (No vector BCD instructions),
+ *
+ *  this implementation only tests for a valid plus sign nibble.
+ *  Otherwise the BCD value is assumed to be negative.
+ *
+ *  |processor| Latency |Throughput|
+ *  |--------:|:-------:|:---------|
+ *  |power8   | 17 - 26 | 2/cycle  |
+ *  |power9   | 5 - 14  | 2/cycle  |
+ *
+ *  @param vra a 128-bit vector treated as signed BCD quadword.
+ *  @return a 128-bit vector bool of all '1's if the sign is negative.
+ *  Otherwise all '0's.
+ */
+static inline vb128_t
+vec_setbool_bcdsq (vBCD_t vra)
+{
+  vb128_t result;
+#if defined (_ARCH_PWR8) && (__GNUC__ > 6)
+  if (__builtin_bcdsub_gt ((vi128_t) vra, (vi128_t) _BCD_CONST_MINUS_ONE, 0))
+    result = (vb128_t) vec_splat_s8 (0);
+  else
+    result = (vb128_t) vec_splat_s8 (-1);
+#else
+  const vui32_t sign_mask = vec_splat_u32(15);
+//  const vui32_t neg_sign = (vui32_t) CONST_VINT128_W(0x0b, 0x0d, 0x0b, 0x0d);
+  const vui32_t plus_sign = (vui32_t) CONST_VINT128_W(0x0a, 0x0c, 0x0e, 0x0f);
+  vui32_t sign_splat;
+  vui32_t sign_code;
+
+  // Replicate the byte containing the sign to words
+  sign_splat = vec_splat ((vui32_t) vra, VEC_W_L);
+  // Apply the code mask
+  sign_code = vec_and (sign_splat, sign_mask);
+  // SIMD compare for match to any positive sign code
+  if (vec_any_eq(sign_code, plus_sign))
+    result = (vb128_t) vec_splat_s8(0);
+  else
+    result = (vb128_t) vec_splat_s8(-1);
+#endif
+  return (result);
+}
+
+/*! \brief Vector Sign bit from Signed BCD Quadword.
+ *
+ *  If the quadword's sign nibble is 0xB or 0xD then return a
+ *  none-zero value. Otherwise if the sign nibble is
+ *  0xA, 0xC, 0xE, or 0xF then return all '0's.
+ *
+ *  /note For _ARCH_PWR7 and earlier (No vector BCD instructions),
+ *  this implementation only tests for a valid minus sign nibble.
+ *  Otherwise the BCD value is assumed to be positive.
+ *
+ *  |processor| Latency |Throughput|
+ *  |--------:|:-------:|:---------|
+ *  |power8   | 15 - 26 | 2/cycle  |
+ *  |power9   | 5 - 14  | 2/cycle  |
+ *
+ *  @param vra a 128-bit vector treated as signed BCD quadword.
+ *  @return a none-zero value if the sign is negative.
+ *  Otherwise return '0's.
+ */
+static inline int
+vec_signbit_bcdsq (vBCD_t vra)
+{
+  int result;
+#if defined (_ARCH_PWR8) && (__GNUC__ > 6)
+  result = __builtin_bcdsub_lt ((vi128_t) vra, (vi128_t) _BCD_CONST_ZERO, 0);
+#else
+  const vui32_t sign_mask = vec_splat_u32(15);
+  const vui32_t minus_sign = (vui32_t) CONST_VINT128_W(0x0b, 0x0d, 0x0b, 0x0d);
+//  const vui32_t plus_sign = (vui32_t) CONST_VINT128_W(0x0a, 0x0c, 0x0e, 0x0f);
```

Comment:
did you want to keep this commented-out line?

Reply:
Removed